--- saltify.py	2022-10-12 21:49:13.410775428 +0000
+++ saltify.py	2022-10-12 21:45:55.817334485 +0000
@@ -29,12 +29,11 @@
 
 try:
     # noinspection PyUnresolvedReferences
-    from impacket.smbconnection import SessionError as smbSessionError
-    from impacket.smb3 import SessionError as smb3SessionError
+    from smbprotocol.exceptions import InternalError as smbSessionError
 
-    HAS_IMPACKET = True
+    HAS_SMB = True
 except ImportError:
-    HAS_IMPACKET = False
+    HAS_SMB = False
 
 try:
     # noinspection PyUnresolvedReferences
@@ -340,8 +339,8 @@
 
         log.debug("Testing Windows authentication method for %s", vm_["name"])
 
-        if not HAS_IMPACKET:
-            log.error("Impacket library not found")
+        if not HAS_SMB:
+            log.error("smbprotocol library not found")
             return False
 
         # Test Windows connection
@@ -360,7 +359,7 @@
             log.debug("Testing SMB protocol for %s", vm_["name"])
             if __utils__["smb.get_conn"](**kwargs) is False:
                 return False
-        except (smbSessionError, smb3SessionError) as exc:
+        except (smbSessionError) as exc:
             log.error("Exception: %s", exc)
             return False
 
